src/presentation/api/routers/subjects.py

```
@@ -0,0 +1,152 @@
+    stmt = select(Subject)
+    if semester_name:
+        stmt = stmt.where(Subject.semester_name == semester_name)
+    if year:
+        stmt = stmt.where(Subject.year == year)
+    if code:
+        stmt = stmt.where(Subject.subject_code == code)
+    return session.exec(stmt).all()
+
+
+@router.post("/", response_model=SubjectRead, status_code=status.HTTP_201_CREATED)
+def create_subject(data: SubjectCreate, session: Session = Depends(get_session)) -> Subject:
```

Ruff (0.14.3)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)